openquake/hazardlib/gsim/projects/acme_2019.py

```diff
@@ -330,33 +379,71 @@ def get_acc_from_disp(self, disp, imt):
 
-    def get_mean_and_stddevs(self, sites, rup, dists, imt, stds_types):
+    def extrapolate_in_PSA(self, sites, rup, dists, imt_high, 
+                           imt_second, stds_types, imt):
+        # get mean at highest imt
+        mean_high, _ = self.gmpe.get_mean_and_stddevs(
+                sites, rup, dists, SA(imt_high), stds_types)
+        
+        mean_low, _ = self.gmpe.get_mean_and_stddevs(
+                sites, rup, dists, SA(imt_second), stds_types)
+
+        delta_y = mean_high - mean_low
+        delta_x = imt_high - imt_second
+
+        mean = mean_high + (delta_y/delta_x) * (imt - imt_high)
+
+        return mean, _
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stds_types, 
+                             extr=True):
+
         nsites = len(sites)
         stddevs = self.get_stddevs(rup.mag, imt, stds_types, nsites)
-
-        # compute corner frequency and capping period
+        
         cornerp = self.get_corner_period(rup.mag)
-        cappingp = self.get_capping_period(cornerp, self.gmpe)
-
-        # apply extrapolation to periods > cappingp
-        if imt.period > cappingp:
-            # compute acceleration at the capping period
+        # capping period only compares 
+        # - highest period with a coefficient
+        # - corner period
+        # - 2.0 if it's bindi 
+        hp, sp = self.get_capping_period(cornerp, self.gmpe, imt)
+
+        # 1 - if imt.period < cornerp, no changes needed
+        if extr and imt.period <= cornerp and imt.period <= hp:
+            mean, _ = self.gmpe.get_mean_and_stddevs(
+                sites, rup, dists, imt, stds_types)
+        # if the period is larger than the corner period but the corner period
+        # is less than the highest period
+        elif extr and imt.period >= cornerp and cornerp <= hp:
             mean, _ = self.gmpe.get_mean_and_stddevs(
-                sites, rup, dists, SA(cappingp), stds_types)
-            # convert to spectral displacement at the capping period
-            disp = self.get_disp_from_acc(mean, cappingp)
+                sites, rup, dists, SA(cornerp), stds_types)
+            disp = self.get_disp_from_acc(mean, cornerp)
             mean = self.get_acc_from_disp(disp, imt.period)
+        # if the corner period is longer than highest and imt is above
+        # highets but below corner
+        elif extr and cornerp > hp and imt.period >= hp and imt.period < cornerp:
+            mean, _ = self.extrapolate_in_PSA(sites, rup, dists,
+                                hp, sp, stds_types, imt.period)
+        elif extr and cornerp > hp and imt.period >= cornerp:
+            mean, _ = self.extrapolate_in_PSA(sites, rup, dists,
+                                hp, sp, stds_types, cornerp)
+            disp = self.get_disp_from_acc(mean, cornerp)
+            mean = self.get_acc_from_disp(disp, imt.period)
+
         else:
+            print('using regular computation! check if meant to extrapolate')
```

Comment:
Looks like a leftover; if you want to display a warning this is not the right place since this function is called millions of times

Reply:
removed this